modules/ppcp-wc-gateway/src/Gateway/ProcessPaymentTrait.php

```diff
@@ -253,13 +253,19 @@ function ( Authorization $authorization ): bool {
 						}
 					}
 
+					// Adds retry counter meta to avoid duplicate invoice id error on consequent tries.
+					$wc_order->update_meta_data( 'ppcp-retry', (int) $wc_order->get_meta( 'ppcp-retry' ) + 1 );
+					$wc_order->save_meta_data();
+
 					$this->session_handler->destroy_session_data();
 					wc_add_notice( $error_message, 'error' );
 
 					return $failure_data;
 				}
 
+				WC()->cart->empty_cart();
```

Comment:
I´ve moved it from `OrderProcessor::process` to here because otherwise there were no chance to retry the payment when the payment is not saved on PayPal.

Reply:
@AlexP11223

In order to prevent subscriptions renewals to fail because no payment stored for customer we are introducing a check after the payment to see if the payment is saved on PayPal or not, if it´s not saved then we set the WC order as failed.

The problem is that if the customer tries again with the same order it gets a **DUPLICATE_INVOICE_ID**, here is how I currently implemented it:
1. POST order with Auth intent
2. PATCH
3. Authorize
Check if payment is saved, if so capture auth order.

The above works, however if point 4 fails, we void the auth order and start again:
1. POST order with Auth intent
2. PATCH
3. Authorize, here we get **DUPLICATE_INVOICE_ID**

The error is because as the PayPal authorization was successful, that order ID is “used”, so we have to increment or otherwise alter the invoice ID.

I´ve implemented the later by prepending counter at the end, so if WC-001 is used, we alter it to WC-001-01.